### DISCUSSION TITLE
Unexpected end of stream in segment

### DISCUSSION THREAD

**Discussion by SashaFokin7** [xlate/staedi#279] [General]
Hello. I'm trying to parse EDI 210 file which contains segment:
`N1*SH*BRISAY~`
and received error: io.xlate.edi.internal.stream.tokenization.EDIException: EDIE005 - Unexpected end of stream in segment N1 at position 9, element 1
stacktrace:
 `io.xlate.edi.internal.stream.tokenization.EDIException: EDIE005 - Unexpected end of stream in segment N1 at position 5, element 1
	at io.xlate.edi.internal.stream.tokenization.Lexer.error(Lexer.java:455)
	at io.xlate.edi.internal.stream.tokenization.Lexer.parse(Lexer.java:304)
	at io.xlate.edi.internal.stream.StaEDIStreamReader.nextEvent(StaEDIStreamReader.java:203)
	at io.xlate.edi.internal.stream.StaEDIStreamReader.next(StaEDIStreamReader.java:236)`
Can you please help?

**Comment by MikeEdgar** [comment]
@SashaFokin7 can you provide the full input (with sensitive info removed)? Would it be possible to post a small reproducer for the error?

**Reply by SashaFokin7** [reply]
input:
```
ISA*00*          *00*          *ZZ*AAAAAAA        *02*AAAAAAA        *220408*0514*U*00401*000000000*0*P*>~
GS*IM*AAAAAAA*AAAAAAA*20220408*051457*16*X*004010~
ST*210*0009~
B3**111111111*111111*PP*L*20220228*12345*PD*20220224*035*SSSS~
N1*SH*BRISAY~
N3*RD SMITH~
N4*WNS*GA*22222~
N1*CN*TEST1~
N3*TEST2~
N4*TEST*TN*11111~
N1*BT*TEST3~
N3*NEW YORK~
N4*FFFFF*II*00000~
LX*1~
L5*1*DESCRIPTION~
L0*1*10.01*LB*5555*G***4*PCS**L~
L1*1*11.11*PH*44444~
L7*1*SEF6*0A****100.0~
SE*17*0009~
GE*1*16~
IEA*1*000000000~
```



@Component
public class Parser {

  private EDIInputFactory factory = EDIInputFactory.newFactory();

  public Edi parse(InputStream source) throws IOException, EDIStreamException {
    try (EDIStreamReader reader = factory.createEDIStreamReader(source)) {
      Edi document = new Edi ();
      while (reader.hasNext()) {
        switch (reader.next()) {
          // case ...
      }
      return document;
    }
  }
}

**Reply by MikeEdgar** [reply]
@SashaFokin7 , I've put together this test and the issue is not occurring. Questions for you:
- What version of staedi are you using? 
- How does your application obtain the `InputStream` that is passed to the `parse` method you gave in your example?
```java
String data = "ISA*00*          *00*          *ZZ*AAAAAAA        *02*AAAAAAA        *220408*0514*U*00401*000000000*0*P*>~\n"
+ "GS*IM*AAAAAAA*AAAAAAA*20220408*051457*16*X*004010~\n"
+ "ST*210*0009~\n"
+ "B3**111111111*111111*PP*L*20220228*12345*PD*20220224*035*SSSS~\n"
+ "N1*SH*BRISAY~\n"
+ "N3*RD SMITH~\n"
+ "N4*WNS*GA*22222~\n"
+ "N1*CN*TEST1~\n"
+ "N3*TEST2~\n"
+ "N4*TEST*TN*11111~\n"
+ "N1*BT*TEST3~\n"
+ "N3*NEW YORK~\n"
+ "N4*FFFFF*II*00000~\n"
+ "LX*1~\n"
+ "L5*1*DESCRIPTION~\n"
+ "L0*1*10.01*LB*5555*G***4*PCS**L~\n"
+ "L1*1*11.11*PH*44444~\n"
+ "L7*1*SEF6*0A****100.0~\n"
+ "SE*17*0009~\n"
+ "GE*1*16~\n"
+ "IEA*1*000000000~";

InputStream source = new ByteArrayInputStream(data.getBytes());
EDIInputFactory factory = EDIInputFactory.newFactory();

try (EDIStreamReader reader = factory.createEDIStreamReader(source)) {
    while (reader.hasNext()) {
        continue;
    }
}
```

**Reply by MikeEdgar** [reply]
@SashaFokin7 were you able to get things working?

**Reply by SashaFokin7** [reply]
@MikeEdgar  sorry for the long reply.
We found the issue on our end. Thank you for help!
